src/backend/base/langflow/components/vectorstores/astradb.py

```diff
@@ -482,43 +491,69 @@ def _map_search_type(self) -> str:
         return "similarity"
 
     def _build_search_args(self):
-        args = {
-            "k": self.number_of_results,
-            "score_threshold": self.search_score_threshold,
-        }
+        query = self.search_input if isinstance(self.search_input, str) and self.search_input.strip() else None
+        search_filter = (
+            {k: v for k, v in self.search_filter.items() if k and v and k.strip()} if self.search_filter else None
+        )
+
+        if query:
+            args = {
+                "query": query,
+                "search_type": self._map_search_type(),
+                "k": self.number_of_results,
+                "score_threshold": self.search_score_threshold,
+            }
+        elif self.advanced_search_filter or search_filter:
+            args = {
+                "n": self.number_of_results,
```

Comment:
Should this be k?

Reply:
No, `metadata_search` uses `n`, not `k`. 

https://python.langchain.com/api_reference/astradb/vectorstores/langchain_astradb.vectorstores.AstraDBVectorStore.html#langchain_astradb.vectorstores.AstraDBVectorStore.metadata_search

Reply:
Hah, you're right indeed. I'm surprised it differs but hey, thats correct :) Approving! Thanks!